File: 프로그래머스/2/42586. 기능개발/기능개발.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int N = progresses.Length;
        var li = new List<int>();
        for (int i=0; i<N; ++i) {
            if ((100-progresses[i]) % speeds[i] == 0) {
                li.Add((100-progresses[i])/speeds[i]);
            }
            else {
                li.Add(((100-progresses[i])/speeds[i])+1);
            }
        }
        bool[] done = new bool[N];
        var ans = new List<int>();
        for (int i=0; i<N; ++i) {
            if (done[i]) {
                continue;
            }
            int cnt=0;
            for (int j=i; j<N; ++j) {
                if ((li[j] <= li[i]) && !done[j]) {
                    cnt++;
                    done[j] = true;
                }
                else {
                    break;
                }
            }
            ans.Add(cnt);
        } 
        return ans.ToArray();      
                
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const answer = [];
    const N = progresses.length;
    
    const remain = [];
    for (let i=0; i<N; ++i) {
        const [p, s] = [progresses[i], speeds[i]];
        remain.push(Math.ceil((100-p) / s));
    }
    
        let deployDay = remain[0];
    let count = 1;

    for (let i = 1; i < N; ++i) {
        if (remain[i] <= deployDay) {
            count++;
        } else {
            answer.push(count);

            deployDay = remain[i];
            count = 1;
        }
    }

    answer.push(count);

    return answer;
    
}